test/modules/test_dangling_dns_detector.py

```diff
@@ -0,0 +1,263 @@
+        mock_aaaa_answer.__iter__.return_value = iter([])
+        mock_resolve.side_effect = [mock_a_answer, mock_aaaa_answer]
+
+        # when
+        result: list[dict[str, Any]] = []
+        self.karton.check_dns_ip_records("norecords.example.com", result)
+
+        # then
+        self.assertFalse(result)
+
+    @patch("dns.resolver.resolve")
+    def test_check_cname_dangling(self, mock_resolve) -> None:  # type: ignore
+        # given
+        cname_record = MagicMock()
+        cname_record.rdtype = rdatatype.CNAME
+        cname_record.target.to_text.return_value = "unregistered.example.com."
+        mock_answer = MagicMock()
+        mock_answer.rrset = [cname_record]
+        mock_answer.__iter__.return_value = iter([cname_record])
+        mock_resolve.side_effect = [mock_answer]
+
+        # when
+        result: list[dict[str, Any]] = []
+        self.karton.check_cname("dangling.example.com", result)
+
+        # then
+        self.assertTrue(len(result) == 1)
+        self.assertTrue(result[0]["record"] == rdatatype.CNAME)
+        self.assertTrue("does not resolve" in result[0]["message"])
+
+    @patch("artemis.modules.dangling_dns_detector.edns_query")
+    @patch("dns.resolver.resolve")
+    def test_check_cname_registered(self, mock_resolve, mock_edns_query) -> None:  # type: ignore
+        # given
+        cname_record = MagicMock()
+        cname_record.rdtype = rdatatype.CNAME
+        cname_record.target.to_text.return_value = "registered.example.com."
+        mock_cname_answer = MagicMock()
+        mock_cname_answer.rrset = [cname_record]
+        mock_cname_answer.__iter__.return_value = iter([cname_record])
+        mock_resolve.side_effect = [mock_cname_answer]
+
+        a_record = MagicMock()
+        a_record.rdtype = rdatatype.A
+        mock_a_answer = MagicMock()
+        mock_a_answer.__iter__.return_value = iter([a_record])
+        mock_edns_query.side_effect = [mock_a_answer, None, None]
+
+        # when
+        result: list[dict[str, Any]] = []
+        self.karton.check_cname("registered.example.com", result)
+
+        # then
+        self.assertFalse(result)
+
+    @patch("dns.resolver.resolve")
+    def test_check_cname_no_cname(self, mock_resolve) -> None:  # type: ignore
+        # given
+        mock_answer = MagicMock()
+        mock_answer.rrset = None
+        mock_answer.__iter__.return_value = iter([])
+        mock_resolve.return_value = mock_answer
+
+        # when
+        result: list[dict[str, Any]] = []
+        self.karton.check_cname("nocname.example.com", result)
+
+        # then
+        self.assertFalse(result)
+
+
+class TestDanglingDnsDetectorIntegration(ArtemisModuleTestCase):
+    karton_class = DanglingDnsDetector  # type: ignore
+
+    def test_cname_dangling_real_domain(self) -> None:
+        # given
+        task = Task(
+            {"type": TaskType.DOMAIN_THAT_MAY_NOT_EXIST.value},
+            payload={"domain": "bad.kadetest.xyz"},
```

Comment:
I'd replace this with something in `cert.pl`, can you specify what config we need so that we can pass it to the admins?